### PR TITLE
[sw,dif,aes] dif aes refactoring

### PR DESCRIPTION
**Pull request by doreis-lowrisc**
This PR Refators the dif_aes. 

To have a better understanding, please look at each commit individually.

### CODE REVIEW

**Comment by doreis-lowrisc** [comment]
@vogelpi could review how I've addressed the first item of the issue #10487?

File: sw/device/lib/dif/dif_aes.c
Comment (vogelpi) [inline sw/device/lib/dif/dif_aes.c:177]: As discussed in person, I think also this conversion from `dif_aes_key_length_t` to `aes_key_field_val_t` could be removed. The same holds for `aes_operation_t` and `aes_operation_field_val_t`. But someone more familiar with DIFs might know why this was introduced in the first place.

Comment (vogelpi) [reply sw/device/lib/dif/dif_aes.c:177]: When doing that, you can also remove the conversion functions `key_to_field` etc. starting at line 111 as well as the references to the issue at lines 123 and 170.

**Review by vogelpi — COMMENTED** [review]
From a hardware perspective, this looks good to me. I've left some comments around the conversion from the `_t` to the `_val_t` types. I am in favor of removing these but the software team might have a different view. Thanks @doreis-lowrisc for working on this!

File: sw/device/lib/dif/dif_aes.c
Comment (doreis-lowrisc) [reply sw/device/lib/dif/dif_aes.c]: Thanks @vogelpi, Done!

Comment (doreis-lowrisc) [reply sw/device/lib/dif/dif_aes.c]: If you are happy, I can move this PR from Draft.

File: sw/device/lib/dif/dif_aes.h
Comment (vogelpi) [inline sw/device/lib/dif/dif_aes.h:199]: This needs to be reverted. the MANUAL_OPERATION field/bit is not one-hot encoded. 0 means automatic operation, 1 means manual operation.

File: sw/device/lib/dif/dif_aes.h
Comment (vogelpi) [inline sw/device/lib/dif/dif_aes.h:204]: This should just be `1` instead of `0x2`.

File: sw/device/lib/dif/dif_aes.h
Comment (doreis-lowrisc) [reply sw/device/lib/dif/dif_aes.h]: Done.

File: sw/device/lib/dif/dif_aes.h
Comment (doreis-lowrisc) [reply sw/device/lib/dif/dif_aes.h]: Done

File: sw/device/lib/dif/dif_aes.h
Comment (vogelpi) [inline sw/device/lib/dif/dif_aes.h:107]: Nit: "The ECB cipher mode doesn't use..."

File: sw/device/lib/dif/dif_aes.h
Comment (vogelpi) [inline sw/device/lib/dif/dif_aes.h:123]: Nit: indentation

File: sw/device/lib/dif/dif_aes.h
Comment (vogelpi) [inline sw/device/lib/dif/dif_aes.h:138]: Nit: indentation

**Review by silvestrst — APPROVED** [review]

**Review by vogelpi — APPROVED** [review]
LGTM! I have just two minor nits.

A couple of notes on the commit messages:
- When refering to issues, use a hashtag + full specifier. In this particular case, you can write `This fixes lowRISC/OpenTitan#10487.` Once the PR is merged, the corresponding issue is automatically closed which is very handy.
- I would add a space after comma in the tags: [sw,dif.aes] -> [sw, dif, aes]
- Use present tense: dif_aes_start refactoring -> Refactor dif_aes_start

File: sw/device/lib/dif/dif_aes.h
Comment (doreis-lowrisc) [reply sw/device/lib/dif/dif_aes.h:263]: Done

File: sw/device/lib/dif/dif_aes.h
Comment (doreis-lowrisc) [reply sw/device/lib/dif/dif_aes.h]: Done

File: sw/device/lib/dif/dif_aes.h
Comment (doreis-lowrisc) [reply sw/device/lib/dif/dif_aes.h:123]: Thanks for noticing, but I think I'll let as it is, since `clang-format` is not complaining.

**Comment by doreis-lowrisc** [comment]
Thanks @silvestrst and @vogelpi for your review.
I've reviewed all the points spotted by you. If you are happy please merge this PR.

**Comment by vogelpi** [comment]
Awesome, thanks @doreis-lowrisc ! I can merge this once CI passes.

File: sw/device/tests/aes_smoketest.c
Comment (vogelpi) [inline sw/device/tests/aes_smoketest.c:126]: @doreis-lowrisc CI is currently failing because `kDifAesOperationEncrypt` and `kDifAesOperationDecrypt` are not properly defined inside `dif_aes.h`. This is now a 2-bit one-hot encoded field. We need:
```c
kDifAesOperationEncrypt = 1;
kDifAesOperationDecrypt = 1 << 1;
```
Related to that, it seems the unit test only tests encryption but not decryption. Should this be corrected as well?

Comment (vogelpi) [reply sw/device/tests/aes_smoketest.c:126]: Perfect, thanks @doreis-lowrisc , I am going to merge this now.

**Pull request merged**
